### ISSUE TITLE
elasticsearch_reader not returning correct number of documents with an es query 

### ISSUE THREAD

**Issue by ciorg** [terascope/teraslice#730]
elasticsearch query returns 37k records, but the same query in the teraslice job returned 111k records.   On inspection teraslice had one slice with the correct number of records, but 4 other slices that also returned additional records.
```
jobe query was: "_exists_:field1 AND NOT _exists_:field2",
```

**Comment by kstaken** [comment]
It was concluded that this issue was caused by 5 slicers trying to process a query that easily fit within a single slice. One slicer had the correct date range and count. The other 4 had different date ranges that extended beyond the time range available in the index and varying record counts that duplicated data from the one correct slice.

**Comment by ciorg** [comment]
Was able to reproduce this error when a job query returns a small data set (320k records)
job:
```
{
    "name": "test#slicer-count-test",
    "lifecycle": "once",
    "workers": 10,
    "assets": [
        "common_processors"
    ],
    "operations": [
        {
            "_op": "elasticsearch_reader",
            "size": 150000,
            "index": "",
            "date_field_name": "",
            "type": "",
            "interval": "auto",
            "connection": "",
            "query": "count:20",
            "full_response": true
        },
        {
            "_op": "noop"
        }
    ],
    "tjm": {
        "cluster": "",
        "version": "0.0.1",
        "job_id": "5cbdecec-fadb-48ac-9fc3-ebfb4a6f414f"
    }
}
```
execution contents:
```
{
    "name": "test#slicer-count-test",
    "lifecycle": "once",
    "analytics": true,
    "max_retries": 3,
    "slicers": 5,
    "workers": 10,
    "operations": [ ...
}
```
the job processed 503k records, but the curl search returned 320k.  5 slicers were used but the default should be 1.

When I re-ran the job with the slicer property set at 1, it processed the correct number of records.

**Comment by ciorg** [comment]
When I set a job to 5 slicers then run the job, then remove the slicer property, update and run the job it still executes with 5 slicers.  If I then set the number of slicers to 1 update and run the job, then remove the slicer property it continues to run with 1 slicer.  Seems like the slicer property is based on the previous time the job was run unless specifically set.  A newly registered job defaults to 1 slicer.
